### PR TITLE
Make `year` Gregorian year and add `julianyear`

### LINKED ISSUE #51: Use Gregorian year instead of Julian one?
Seeing 92feca4ec665d129d40aa31dbd70da34414cb483, I'm a bit puzzled as to why the Julian year was used, and not the Gregorian one, which is both much more widespread (https://en.wikipedia.org/wiki/Adoption_of_the_Gregorian_calendar#Present_situation) and more accurate. Is there a reason for that?

Naturally, if the year were to be changed, the month should be too.

Alternatives I considered:

- maybe we can have a `julianyear` or something like that, and change `year` to be a Gregorian year (or vice versa)?

### PR DESCRIPTION
**Pull request by triallax**
Fixes #51.

This is a breaking chance that must be announced.

### CODE REVIEW

File: src/Data/Units/Time.purs [hunk 1]
```diff
@@ -22,10 +22,14 @@ day = makeNonStandard "day" "d" (24.0 * 3600.0) second
...
 
--- | Unit of time, *1 year = 365.25 days* (Julian year).
+-- | Unit of time, *1 year = 365.2425 days* (Gregorian year).
 year ∷ DerivedUnit
-year = makeNonStandard "year" "year" (365.25 * 24.0 * 3600.0) second
+year = makeNonStandard "year" "year" (365.2425 * 24.0 * 3600.0) second
+
+-- | Unit of time, *1 julianyear = 365.25 days*.
+julianyear ∷ DerivedUnit
+julianyear = makeNonStandard "julianyear" "julianyear" (365.25 * 24.0 * 3600.0) second
```
Comment (triallax) [inline src/Data/Units/Time.purs]: Should the name of the unit be `juilanyear` or `julianYear`? I could find no consistent choice regarding units with compound names; light year has a name of `lightyear`, while nautical mile is `nauticalMile`.

Also, should the long name of the unit be `julianyear` or julian year? `nauticalMile` has `nautical mile` as a long name, while `lightyear` has no space in its long name.

Sources:
https://github.com/sharkdp/purescript-quantities/blob/050f3fc06b2c9b7d4a1feb9cae7b3f073bbd410b/src/Data/Units/Nautical.purs#L11
https://github.com/sharkdp/purescript-quantities/blob/050f3fc06b2c9b7d4a1feb9cae7b3f073bbd410b/src/Data/Units/Astronomical.purs#L13

Comment (sharkdp) [reply src/Data/Units/Time.purs]: > Should the name of the unit be `juilanyear` or `julianYear`? I could find no consistent choice regarding units with compound names; light year has a name of `lightyear`, while nautical mile is `nauticalMile`.

Hm. "lightyear" can really be spelled that way (https://en.wikipedia.org/wiki/Light-year). For everything else, I would go with the camelCase variant. So I would propose we use "julianYear".

> Also, should the long name of the unit be `julianyear` or julian year? `nauticalMile` has `nautical mile` as a long name, while `lightyear` has no space in its long name.

Let's use the space for the camelCase variants. So "julian year".

**Review by sharkdp — APPROVED** [review]
Very nice! Approving. Doing the naming changes for the new/old Julian year would be great.

File: src/Data/Units/Time.purs [hunk 2]
```diff
@@ -22,10 +22,14 @@ day = makeNonStandard "day" "d" (24.0 * 3600.0) second
...
 
--- | Unit of time, *1 year = 365.25 days* (Julian year).
+-- | Unit of time, *1 year = 365.2425 days* (Gregorian year).
 year ∷ DerivedUnit
-year = makeNonStandard "year" "year" (365.25 * 24.0 * 3600.0) second
+year = makeNonStandard "year" "year" (365.2425 * 24.0 * 3600.0) second
+
+-- | Unit of time, *1 julianYear = 365.25 days*.
+julianYear ∷ DerivedUnit
+julianYear = makeNonStandard "julian year" "julianYear" (365.25 * 24.0 * 3600.0) second
```
Comment (triallax) [inline src/Data/Units/Time.purs:35]: Is this what you had in mind?

Comment (sharkdp) [reply src/Data/Units/Time.purs:35]: Exactly.

**Review by sharkdp — APPROVED** [review]

**Pull request merged**